src/Transport.ts

```diff
@@ -178,6 +178,24 @@ export class Transport {
     this.ws.send(this.buildMsg(['asset', '', '', _uri]));
   }
 
+  /**
+   * Send a message through the websocket, if the connection status allows it.
+   *
+   * @param msg The message to send. It consists of four parts:
+   *   1. Operation
+   *   2. Topic name
+   *   3. Message type
+   *   4. Payload
+   */
+  public sendMessage(msg: string[]): void {
+    // Only send the message when the connection allows it.
+    const connectionStatus = this.status$.getValue();
+    if (connectionStatus === 'disconnected' || connectionStatus === 'error') {
+      return;
+    }
+    this.ws.send(this.buildMsg(msg));
```

Comment:
I feel like we should validate that `msg` has 4 parts here, otherwise we run the risk of creating a message that the backend can't understand.

Reply:
Relatedly, we could also change it so that all of the other uses of `this.ws.send()` instead use `this.sendMessage`, which would make the whole thing more consistent.

Reply:
That's a great suggestion. I'll try to make the send more robust.

Reply:
Addressed in https://github.com/gazebo-web/gzweb/pull/24/commits/dd3d12aecdac1d7cba96f8b7002e0787f4bdf8ea.

- Now Transport uses `this.sendMessage` instead of `this.ws.send`.
- sendMessage method now checks:
  - That the message consists on four parts.
  - Connection is not error.
  - Connection is not disconnected. Note: Some messages need to be sent to achieve the Connected status (auth, for example), so we need to take those into account.
  - Log errors, for better visibility.

Let me know what do you think @clalancette